src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -146,9 +146,27 @@ SettingsLoader::SettingsLoader(const std::string_view& userJSON, const std::stri
     if (const auto sources = userSettings.globals->DisabledProfileSources())
     {
         _ignoredNamespaces.reserve(sources.Size());
-        for (const auto& id : sources)
+        for (auto&& id : sources)
         {
-            _ignoredNamespaces.emplace(id);
+            _ignoredNamespaces.emplace(std::move(id));
+        }
+    }
+
+    // Apply DisabledProfileSources policy setting. Pick whatever policy is set first.
+    // In most cases HKCU settings take precedence over HKLM settings, but the inverse is true for policies.
+    for (const auto key : { HKEY_LOCAL_MACHINE, HKEY_CURRENT_USER })
+    {
+        wchar_t buffer[512]; // "640K ought to be enough for anyone"
+        DWORD bufferSize = sizeof(buffer);
+        if (RegGetValueW(key, LR"(Software\Policies\Microsoft\Windows\Terminal)", L"DisabledProfileSources", RRF_RT_REG_MULTI_SZ, nullptr, buffer, &bufferSize) == 0)

```

Comment:
```suggestion
        if (RegGetValueW(key, LR"(Software\Policies\Microsoft\Windows\WondowsTerminal)", L"DisabledProfileSources", RRF_RT_REG_MULTI_SZ, nullptr, buffer, &bufferSize) == 0)
```

RDP policies using "Terminal" to. In a slightly different place but to avoid confusion you should change the registry key.

Reply:
Or `Microsoft\WindowsTerminal` is possible too.

Reply:
OMG thank you so much for noticing this! That would've been quite the headache otherwise...
I'll go with `Microsoft\WindowsTerminal` as this mirrors `Microsoft\Windows Defender`. I wonder if we should also use a space like Defender does... 🤔

Reply:
Honestly let’s go with the space

Reply:
The space decision is on you. Both will work.